packages/sanity/src/_internal/cli/actions/graphql/gen3/generateTypeQueries.ts

```diff
@@ -103,8 +106,17 @@ export function generateTypeQueries(
           isFieldFilter: false,
         },
       ],
+      ...getDeprecation(type),
     })
   })
 
   return queries
 }
+
+function getDeprecation(type: ConvertedType): Partial<Deprecation> {
+  return type[internal]?.deprecationReason
+    ? {
+        deprecationReason: type[internal].deprecationReason,
+      }
+    : {}
```

Comment:
❓could this be just undefined instead of an empty object getting added to every type. Maybe in the above lines the spread could be conditional

Reply:
We could, but we'd need to repeat that conditional every time `getDeprecation` is called. That's only twice in this file, but in `packages/sanity/src/_internal/cli/actions/graphql/extractFromSanitySchema.ts`, we have an analogous `getDeprecation` function that is called many more times.

I might be missing something, but what are the benefits would be of returning `undefined` and repeating the conditional logic?

Reply:
The only reason to repeat the conditional is because if you look at the snapshot it adds an empty object. 

```
Object {
      "fields": Array [
	@@ -790,6 +794,7 @@ Object {
      "name": "CdrPersonReference",
      "originalName": "cdrPersonReference",
      "type": "Object",
      Symbol(internal): Object {},
    },
```

which is not a falsy value, personally I would not anything if the field is not deprecated, empty objects might cause unknown issues

Reply:
The `internal` key is intended as a generic container for any data we need to reference during the API generation process, but do not wish to include in then generated API definition. In this case, it's used when we generate the queries to send to Bowtie. It would be incorrect to send Bowtie the deprecation on the document type itself.